cockatrice/src/abstractcounter.cpp

```diff
@@ -27,12 +27,14 @@ AbstractCounter::AbstractCounter(Player *_player,
       useNameForShortcut(_useNameForShortcut), hovered(false), aDec(nullptr), aInc(nullptr), dialogSemaphore(false),
       deleteAfterDialog(false), shownInCounterArea(_shownInCounterArea), game(_game)
 {
+    QString displayName = QString(_name);
```

Comment:
Construct this as close to the display as possible. In this case, right before you use it, inside the if statement